src/seedu/addressbook/data/Tagging.java

```diff
@@ -0,0 +1,27 @@
+package seedu.addressbook.data;
+
+import java.util.ArrayList;
+/*
+ * Association class that records all addition and deletion of tags during a session
+ */
+public class Tagging {
+	private static final String ADD_COMMAND = "add";
+	private static final String DEL_COMMAND = "delete";
+	private ArrayList<String> allTagHistory;
```

Comment:
Based on the class diagram given in the exercise, the list of `Tagging` objects should not be in this class. If it was in this class, the diagram would be like this:
![image](https://cloud.githubusercontent.com/assets/1673303/18409597/69111e4e-777d-11e6-92bf-9914c825bd13.png)